ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -727,18 +683,7 @@ int room_duplicate(int indx, bool ass, int assroom)
   rm->views_enabled = copyrm->views_enabled;
   rm->createcode = copyrm->createcode;
   rm->precreatecode = copyrm->precreatecode;
-
-  enigma::inst *newinst = new enigma::inst[copyrm->instancecount];
-  rm->instances = newinst;
-  rm->instancecount = copyrm->instancecount;
-  for (int i = 0; i < rm->instancecount; i++)
-    rm->instances[i] = copyrm->instances[i];
-
-  enigma::tile *newtiles = new enigma::tile[copyrm->tilecount];
-  rm->tiles = newtiles;
-  rm->tilecount = copyrm->tilecount;
-  for (int i = 0; i < rm->tilecount; i++)
-    rm->tiles[i] = copyrm->tiles[i];
+  rm->tiles = copyrm->tiles;
```

Comment:
You copy tiles to the duplicate room but not the instances. Why don't you?

Reply:
Also, correct me if I'm wrong, but you're now making a shallow copy where we actually want to make a deep copy.

Reply:
Ok: 2a6feea5e8cda61b4d00b061121adf82fe0f7594

I fixed it and it seems ok.